.github/workflows/ci.yml

```
@@ -4,44 +4,55 @@ on:
   - push
   - pull_request
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions: null
```

actionlint (1.7.4)

"null" is invalid for permission for all the scopes. available values are "read-all" and "write-all"

(permissions)